part1/stages/Check-GPU

```diff
@@ -56,4 +56,102 @@ if [ ${GPU_OK} -ne 0 ] ; then
     echo "Ignoring unrecognised graphics device.">&2
 fi
 
+get_vga_pci_info()
+{
+    lspci -m | grep "VGA"
+}
+
+get_pci_id()
+{
+    echo $1 | awk '{print $1}'
+}
+
+get_pci_bus()
+{
+    echo $1 | cut -d: -f 1
+}
+
+get_onboard_pci_id()
+{
+    OLD_IFS=$IFS
+    IFS="
+"
+    for vga_dev in $(get_vga_pci_info); do
+        vga_id="$(get_pci_id $vga_dev)"
+        if [ "$(get_pci_bus $vga_id)" -eq "00" ]; then
+            echo "$vga_id"
+            break
+        fi
+    done
+    IFS=${OLD_IFS}
+}
+
+# Sourced from https://stackoverflow.com/a/8811800
+# BusyBox capable substring search
+# Try to remove the search substring from the search target string
+# If the stripped string is the same as the original target, substing was not found
+contains() {
+    string="$1"
+    substring="$2"
+    if test "${string#*$substring}" != "$string"
+    then
+        return 0    # $substring is in $string
+    else
+        return 1    # $substring is not in $string
+    fi
+}
+
+check_vga()
+{
+    OLD_IFS=$IFS
+    IFS="
+"
+    oboard_pci="$(get_onboard_pci_id)"
+    for boot_vga in $(find /sys/ | grep boot_vga); do
+        if contains "$boot_vga" "$oboard_pci"; then
+            [ "$(cat $boot_vga)" -eq "1" ] || ret=1
```

Comment:
It looks like you can `return` directly when you found it (see previous comment for simplification).

Reply:
I didn't return here so that IFS could be reset to its old value